docker/Dockerfile

```diff
@@ -2,8 +2,8 @@ ARG ALPINE_VERSION=3.15
 ARG BUILDPLATFORM=linux/amd64
 FROM --platform=$BUILDPLATFORM ${DOCKER_REGISTRY:+$DOCKER_REGISTRY/}golang:1.17-alpine${ALPINE_VERSION} as go-builder
 
-ARG TARGETOS=linux
-ARG TARGETARCH=amd64
+ARG TARGETOS
+ARG TARGETARCH
```

Comment:
Don't you also need to add `--platform=$TARGETPLATFORM` on the `FROM alpine:3.15 AS postgres-operator` line 20 below? 

```Dockerfile
FROM --platform=$TARGETPLATFORM alpine:3.15 AS postgres-operator
```

```

Reply:
Hi @mprimeaux, so sorry for the delay on this. Unfortunately as I mentioned in the corresponding [posts](https://github.com/zalando/postgres-operator/issues/1940#issuecomment-1318864710) this was used in a demo and I honestly can't remember! But on looking at this, it does seem quite sensible. I think I got away with it because I was building from arm to arm maybe?

Reply:
@abangser No worries. Appreciate you taking time to reply.  

Yes, I suspect you got away with it because you were on an ARM64 CPU. I'm not sure if you build using nerdctl or docker. If using docker then `docker buildx` would be used instead. If using nerdctl then you can use [this](https://github.com/containerd/nerdctl/blob/main/docs/multi-platform.md).

Reply:
Yea I was using docker and specifically `buildx` in the new Make target.

Do think the additional `--platform` would be needed for proper multi platform builds 👍

Reply:
@abangser Yes, since you're using `docker buildx`, you'll need to run the following:

```shell
docker buildx create --platform linux/arm64, linux/amd64, linux/arm/v8 --use
docker buildx build --platform linux/arm64, linux/amd64, linux/arm/v8 --push
```

This will build the container images for the specified platforms and then push to the target OCI registry